Mathlib/Topology/Order/ScottTopology.lean

```diff
@@ -0,0 +1,439 @@
+-- porting note: removed @[simp] to make linter happy
+lemma ofScott_inj {a b : WithScottTopology α} : ofScott a = ofScott b ↔ a = b :=
+Iff.rfl
+
+/-- A recursor for `WithScottTopology`. Use as `induction x using WithScottTopology.rec`. -/
+protected def rec {β : WithScottTopology α → Sort _}
+  (h : ∀ a, β (toScott a)) : ∀ a, β a := fun a => h (ofScott a)
+
+
+instance [Nonempty α] : Nonempty (WithScottTopology α) := ‹Nonempty α›
+instance [Inhabited α] : Inhabited (WithScottTopology α) := ‹Inhabited α›
+
+variable [Preorder α]
+
+instance : Preorder (WithScottTopology α) := ‹Preorder α›
+
+instance : TopologicalSpace (WithScottTopology α) := ScottTopology'
+
+end WithScottTopology
+
+/--
+The Scott topology is defined as the join of the topology of upper sets and the topological space
+where a set `u` is open if, when the least upper bound of a directed set `d` lies in `u` then there
+is a tail of `d` which is a subset of `u`.
+-/
+class ScottTopology (α : Type _) [t : TopologicalSpace α] [Preorder α] : Prop where
+  topology_eq_ScottTopology : t = ScottTopology'
+
+instance [Preorder α] : ScottTopology (WithScottTopology α) :=
+  ⟨rfl⟩
+
+namespace ScottTopology
+
+section preorder
+
+variable [Preorder α]
+
+lemma isOpen_iff_upper_and_Scott_Hausdorff_Open' {u : Set α} :
+  IsOpen (WithScottTopology.ofScott ⁻¹' u) ↔ (IsUpperSet u ∧ ScottHausdorffTopology.IsOpen u) :=
+by rfl
+
+variable [TopologicalSpace α] [ScottTopology α]
+
+variable (α)
+
+lemma topology_eq : ‹_› = ScottTopology' := topology_eq_ScottTopology
+
+variable {α}
+
+/-- If `α` is equipped with the Scott topology, then it is homeomorphic to `WithScottTopology α`.
+-/
+def withScottTopologyHomeomorph : WithScottTopology α ≃ₜ α :=
+  WithScottTopology.ofScott.toHomeomorphOfInducing ⟨by erw [topology_eq α, induced_id]; rfl⟩
+
+end preorder
+
+section morphisms
+
+variable [Preorder α] [Preorder β]
+
+open TopologicalSpace
+
+lemma upperSetTopology_coinduced' {t₁ : TopologicalSpace α} [UpperSetTopology α]
+  (hf : Monotone f) : coinduced f t₁ ≤ @ScottTopology' β _ := by
+  rw [ScottTopology']
+  apply le_sup_of_le_left
+  rwa [← continuous_iff_coinduced_le, ← UpperSetTopology.monotone_iff_continuous]
+
+lemma Monotone_coinduced {t₁ : TopologicalSpace α} [UpperSetTopology α]
+  {t₂ : TopologicalSpace β} [ScottTopology β] (hf : Monotone f) : coinduced f t₁ ≤ t₂ := by
+  rw [ScottTopology.topology_eq β]
+  apply upperSetTopology_coinduced' hf
+
+lemma Monotone_continuous {t₁ : TopologicalSpace α} [UpperSetTopology α]
+  {t₂ : TopologicalSpace β} [ScottTopology β] {f : α → β} (hf : Monotone f)  : Continuous f := by
+  rw [continuous_iff_coinduced_le]
+  apply Monotone_coinduced hf
+
+
+
+end morphisms
+
+section preorder
+
+variable [Preorder α] [TopologicalSpace α] [ScottTopology α]
+
+lemma isOpen_iff_upper_and_Scott_Hausdorff_Open {u : Set α} : IsOpen u
+↔ (IsUpperSet u ∧ ScottHausdorffTopology.IsOpen u) := by erw [topology_eq α]; rfl
```

Comment:
Indentation is wrong

Reply:
I've adjusted the indentation on several results, hopefully right now.